indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -1018,4 +1022,37 @@ List<Task> getTasks()
       giant.unlock();
     }
   }
+
+  void validateTaskPayload(Task task)
+  {
+    try {
+      String payload = passwordRedactingMapper.writeValueAsString(task);
+      if (payload.length() > TASK_SIZE_WARNING_THRESHOLD) {
+        log.warn("Received a large task payload [%s] with id [%s] and datasource [%s]" +
+                " There may be downstream issues caused by managing this large payload." +
```

Comment:
Please update the messaging based on the older suggestion.

Reply:
Also, should we continue to warn if someone has explicitly set the `maxTaskPayloadSize` to be larger than the `TASK_SIZE_WARNING_THRESHOLD`?  I guess it makes sense to always give out a warning to encourage users/operators to reduce the task payload sizes.